crates/primitives/src/genesis.rs

```diff
@@ -165,7 +165,7 @@ impl Encodable for GenesisAccount {
                     return EMPTY_ROOT
                 }
                 let storage_values =
-                    storage.iter().filter(|(_k, &v)| v != KECCAK_EMPTY).map(|(&k, v)| {
+                    storage.iter().filter(|(_k, &v)| v != H256::zero()).map(|(&k, v)| {
```

Comment:
this checks, for 0x0000 and not keccak([]),
which I guess is the right fix because be want to skip `0` values, right?

Reply:
Correct

Reply:
So should we change this filter to only happen after spurious dragon? The state clearing EIPs seem to have to do with deleting entire accounts, not zero-value storage. Geth will not recognize zero value storage when determining the genesis state root, and we do need to be consistent with that behavior, so we can't just remove the filter entirely

Reply:
Ah, you are right, my bad.